backend/icpy/services/whatsapp/access_control.py

```
@@ -0,0 +1,245 @@
+    async def load(self):
+        """Load allowlist and pending pairings from disk."""
+        async with self._lock:
+            allow_path = self.data_dir / "allowlist.json"
+            pending_path = self.data_dir / "pending_pairings.json"
+
+            if allow_path.exists():
+                try:
+                    data = json.loads(allow_path.read_text())
+                    self.allowlist = set(data if isinstance(data, list) else [])
+                    logger.info(f"Loaded {len(self.allowlist)} allowed numbers")
+                except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)